File: problem/0628_Maximum Product of Three Numbers.js
// Given an integer array nums, find three numbers whose product is maximum and return the maximum product.

/**
 * @param {number[]} nums
 * @return {number}
 */
var maximumProduct = function(nums) {
  nums.sort((a, b)=> b-a)
  let pos = nums[0]*nums[1]*nums[2]
  let neg = nums[0]*nums[nums.length-1]*nums[nums.length-2]
  return Math.max(pos, neg)
};

console.log(maximumProduct([-100,-98,-1,2,3,4])) //39200 
console.log(maximumProduct([-1,-2,-3])) //-6

File: c#-solution/0628. Maximum Product of Three Numbers.cs
public class Solution {
    public int MaximumProduct(int[] nums) {
        Array.Sort(nums);
        int num1 = 0;
        int num2 = 0;
        num1 = nums[0] * nums[1] * nums[nums.Length-1];
        num2 = nums[nums.Length-1] * nums[nums.Length-2] * nums[nums.Length-3];

        if(num1 > num2){
            return num1;
        } else {
            return num2;
        };
    }
}